provider/datagen/src/transform/cldr/units/mod.rs

```diff
@@ -40,101 +90,94 @@ impl DataProvider<UnitsConstantsV1Marker> for crate::DatagenProvider {
+                        non_scientific_constant
+                            .clean_num
+                            .extend(clean_constant.clean_den.clone());
+                        non_scientific_constant
+                            .clean_den
+                            .extend(clean_constant.clean_num.clone());
+
+                        updated = true;
                     } else {
-                        temp_den.push(den_str.clone());
+                        non_scientific_constant.non_scientific_den.push_back(den);
                     }
                 }
-
-                constants_with_constants_map_replaceable
-                    .insert(cons_name, (temp_num, temp_den, temp_constant_exactness));
             }
 
-            constants_map_in_str_form = constants_with_constants_map_replaceable;
-
-            if !has_internal_constants {
-                break;
+            if non_scientific_constant.is_free_of_non_scientific() {
+                clean_constants_map.insert(constant_key, non_scientific_constant);
+            } else {
+                constants_with_non_scientific.push_back((constant_key, non_scientific_constant));
             }
-        }
 
-        if max_depth_reached >= maximum_depth {
-            return Err(DataError::custom(
-                "Maximum depth reached while parsing constants. \
-                This is likely due to a circular dependency in the constants. \
-                Note: If the depth was increased, you may need to increase the maximum depth in the code.",
-            ));
+            count = if !updated { count + 1 } else { 0 };
+            if count > constants_with_non_scientific.len() {
+                return Err(DataError::custom(
+                    "An Infinite loop was detected in the CLDR constants data!",
+                ));
+            }
         }
 
-        // Transforming the `constants_map_in_str_form` map into a ZeroMap of `ConstantValue`.
-        // This is done by converting the numerator and denominator slices into a fraction,
-        // and then transforming the fraction into a `ConstantValue`.
+        // Convert `clean_constants_map` into a ZeroMap of `ConstantValue`.
+        // This involves transforming the numerator and denominator slices into a fraction,
+        // and subsequently converting the fraction into a `ConstantValue`.
         let constants_map = ZeroMap::from_iter(
-            constants_map_in_str_form
+            clean_constants_map
                 .into_iter()
-                .map(|(cons_name, (num, den, constant_exactness))| {
-                    let value = convert_slices_to_fraction(&num, &den)?;
+                .map(|(cons_name, constant)| {
+                    let value = convert_slices_to_fraction(
+                        &constant
+                            .clean_num
+                            .iter()
+                            .map(|s| s.to_string())
```

Comment:
nit: there's no need for this to be `String`, `&str` should work fine.

Reply:
done